UNWIND with WHERE generate failed gremlin

Sample query

```
UNWIND {data} as data
WITH data
WHERE data.person.name IS NULL
RETURN data
```

Where data is 
```
{
  data: [ 
    { 
      person: { name: "a" }
    }
  ]
}
```
Produce next gremlin
```
g.inject('  cypher.start')
.choose(__.constant([[person: [name: 'a']]]), __.constant([[person: [name: 'a']]]), __.constant('  cypher.null'))
.unfold().as('data')
.project('data').by(__.identity())
.where(__.project('  GENERATED1', '  GENERATED2')
.by(__.select('data'))
.by(__.constant('person'))
.select(values)
.map(cypherContainerIndex())
.is(neq('  cypher.null'))
.hasNot('name'))
.select('data').as('data').project('data')
.by(__.identity())
```

And raise exception when exectuted. I check on TinkerGraph 

```
Caused by: java.lang.ClassCastException: java.util.LinkedHashMap cannot be cast to org.apache.tinkerpop.gremlin.structure.Element
	at org.apache.tinkerpop.gremlin.process.traversal.step.map.PropertiesStep.flatMap(PropertiesStep.java:50) ~[gremlin-core-3.3.4.jar:3.3.4]
	at org.apache.tinkerpop.gremlin.process.traversal.step.map.FlatMapStep.processNextStart(FlatMapStep.java:49) ~[gremlin-core-3.3.4.jar:3.3.4]
	at org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext(AbstractStep.java:143) ~[gremlin-core-3.3.4.jar:3.3.4]
	at org.apache.tinkerpop.gremlin.process.traversal.util.DefaultTraversal.hasNext(DefaultTraversal.java:192) ~[gremlin-core-3.3.4.jar:3.3.4]
	at org.apache.tinkerpop.gremlin.process.traversal.util.TraversalUtil.test(TraversalUtil.java:84) ~[gremlin-core-3.3.4.jar:3.3.4]
	at org.apache.tinkerpop.gremlin.process.traversal.step.filter.NotStep.filter(NotStep.java:46) ~[gremlin-core-3.3.4.jar:3.3.4]
	at org.apache.tinkerpop.gremlin.process.traversal.step.filter.FilterStep.processNextStart(FilterStep.java:38) ~[gremlin-core-3.3.4.jar:3.3.4]
```

--- a/testware/integration-tests/src/test/java/org/opencypher/gremlin/queries/UnwindTest.java
+++ b/testware/integration-tests/src/test/java/org/opencypher/gremlin/queries/UnwindTest.java
@@ -15,7 +15,10 @@
  */
 package org.opencypher.gremlin.queries;
 
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonMap;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.opencypher.gremlin.test.TestCommons.parameterMap;
 
 import java.util.List;
 import java.util.Map;
@@ -32,6 +35,10 @@ public class UnwindTest {
 
     private List<Map<String, Object>> submitAndGet(String cypher) {
         return gremlinServer.cypherGremlinClient().submit(cypher).all();
+    }
+
+    private List<Map<String, Object>> submitAndGet(String cypher, Object... parameters) {
+        return gremlinServer.cypherGremlinClient().submit(cypher, parameterMap(parameters)).all();
     }
 
     @Test
@@ -179,4 +186,25 @@ public class UnwindTest {
             .isEmpty();
     }
 
+    @Test
+    public void unwindNestedParams() {
+        Map<String, Map<String, String>> noName = singletonMap("person",
+            singletonMap("lastname", "b")
+        );
+        Map<String, Map<String, String>> withName = singletonMap("person",
+            singletonMap("name", "a")
+        );
+
+        List<Map<String, Object>> results = submitAndGet(
+            "UNWIND {data} as data\n" +
+                "WITH data\n" +
+                "WHERE data.person.name IS NULL\n" +
+                "RETURN data",
+            "data", asList(withName, noName)
+        );
+
+        assertThat(results)
+            .extracting("data")
+            .containsExactly(noName);
+    }
 }

--- a/translation/src/main/scala/org/opencypher/gremlin/translation/walker/WhereWalker.scala
+++ b/translation/src/main/scala/org/opencypher/gremlin/translation/walker/WhereWalker.scala
@@ -102,7 +102,7 @@ private class WhereWalker[T, P](context: WalkerContext[T, P], g: GremlinSteps[T,
           walkExpression(expr).flatMap(extractStep(keyName))
         }.getOrElse {
           val key = StringLiteral(keyName)(InputPosition.NONE)
-          asList(expr, key).map(CustomFunction.cypherContainerIndex()).is(p.neq(NULL))
+          asList(expr, key).map(CustomFunction.cypherContainerIndex())
         }
 
       case ContainerIndex(expr, idx) =>
@@ -121,7 +121,7 @@ private class WhereWalker[T, P](context: WalkerContext[T, P], g: GremlinSteps[T,
 
       case IsNull(expr) =>
         expr match {
-          case Property(propertyExpr, PropertyKeyName(keyName: String)) =>
+          case Property(propertyExpr, PropertyKeyName(keyName: String)) if isElement(propertyExpr, context) =>
             walkExpression(propertyExpr).hasNot(keyName)
           case _ =>
             walkExpression(expr).is(p.isEq(NULL))

